app/controllers/api/v2/hosts_bulk_actions_controller.rb

```diff
@@ -63,6 +63,17 @@ def build
         end
       end
 
+      api :PUT, "/hosts/bulk/reassign_hostgroups", N_("Reassign hostgroups")
+      param_group :bulk_host_ids
+      param :hostgroup_id, :number, :required => true, :desc => N_("ID of the hostgroup to reassign the hosts to")
+      def reassign_hostgroup
+        hostgroup = Hostgroup.find(params[:hostgroup_id])
+        BulkHostsManager.new(hosts: @hosts).reassign_hostgroups(hostgroup)
+        process_response(true, { :message => n_("Reassigned %s host to hostgroup %s",
+          "Reassigned %s hosts to hostgroup %s",
+          @hosts.count) % [@hosts.count, hostgroup.name] })
```

Comment:
Consider a host already had the hostgroup assigned. You still count it as reassigned, but is that really true?

Reply:
We have this text: "Some hosts may already be in your chosen host group." in the modal description. Do we have similar pattern when editing other records cause I think just editing and saving gives the notice of record saved.

Reply:
I suppose "assigned" is more technically accurate than "reassigned" because it can include "assigned idempotently," but I'd vote for keeping "reassigned" since that's the goal of the user here.